Configuring Okta SSO in Directus App (#17)
directus-community/guest-authoring: directus-okta-sso/index.md

---

> title: 'Configuring Okta SSO in Directus App'

Reviewer: Oh one more - I would title this Configuring Okta SSO For a Directus Project

description: 'How to integrate Okta SSO with the SAML platform in a Directus application'
author:
  name: 'Maksym Yakubov'

>   avatar_file_name: 'author.jpg'

Author: Yes, of course you can crop the photo as needed for the blog

---

> ## Introduction

Reviewer 2: Yes, of course you can crop the photo as needed for the blog

|------------|-------------|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| email      | Basic       | user.email     | An important field by which Directus will match the email from our Okta account with the email in our app                                                                        |
| first_name | Basic       | user.firstName | The field by which Directus will match the first name from the Okta account with the first name in our app. Add it if we want the user's first name to be pre-filled in the app. |
| last_name  | Basic       | user.lastName  | The field by which Directus will match the last name from the Okta account with the last name in our app. Add it if we want the user's last name to be pre-filled in the app.    |

Let's leave the rest of the parameters at this step as defaults. As a result, we will get parameters like these:

> ![Configure general SAML settings](configure-saml-screen-settings.png)

Reviewer: I'd move this up to just below the first table

![Configure SAML attributes](configure-saml-screen-attributes.png)

> Click **Next** and go to the third and final screen for the Okta app settings.
> 
> ### Feedback
> Our setup is complete, and we are prompted to leave feedback or submit the app to Okta for review. Unless we want to spend any more time here, select **I'm an Okta customer adding an internal app** and, without filling anything in, click **Finish**.
> 
> ![Process feedback screen](feedback-screen.png)

Reviewer: ```suggestion
Click **Next**, select I'm an **I'm an Okta customer adding an internal app** when asked for feedback and finish the app creation wizard. 
```

Kill the whole Feedback section in my opinion as it doesn't impact on what you're trying to teach.

### Add Users

> Now, we can choose which people from our organization can access Directus through Okta SSO. Go to the **Assignments** tab in the settings of our Okta application.

Reviewer: ```suggestion
Choose which people from our organization can access Directus through Okta SSO. Go to the **Assignments** tab in the settings of our Okta application.
```

I've noticed `Now, ` be present a few times. This is more spoken than it is written. Always try and lead with instruction when technical writing for more concise pieces.

![User assignments screen](user-assignments.png)

Click **Assign**->**Assign to People**.

![Assign to people action button](assign-to-people-action.png)

Click **Assign** next to the desired user.

![A list of people to assign](assign-people-modal.png)

We can also change the username, but we will leave the default. Click **Save and Go Back**.

> ![Save the assigned user](save-assigned-user.png)

Reviewer: This is good, but I don't think it's complex enough to demand screenshots. I'd remove them and turn this into a numbered list. 

I'd also add the confirmation to the end of the list. Something like 'The selected user now has permission to access Directus.'

```
:::How to add Users to Organisation

Go to the People page. To do this, click **Directory**->**People** in the sidebar.

![People page](people-page.png)

Click **Add Person**, and enter the user's personal data in the modal window that appears. Then click **Save**.

![Add new person to the organisation](add-person-modal.png)

Great, the user will receive an email with further instructions and appear in our organisation's list of users.

> ```

Reviewer: I would also remove this. Not our job to teach people how to use Okta.

### Result

> Congrats! We have completed our Okta app.

Reviewer: I would remove this line. I thought we were done on first read. 😆

> All we need to do now is go to the **Sign On** tab in our app(**Applications**->**Applications**->**My Directus app**->**Sign On**) and copy the **Metadata URL**. We will need this a bit later.

Reviewer: ```suggestion
Go to the **Sign On** tab in our Okta app(**Applications** -> **Applications** -> **My Directus app** -> **Sign On**) and copy the **Metadata URL**. We will need this to configure our Directus project.
```

![Final screen with the Metadata URL](final-screen.png)

## Configuring SSO in the Directus application

> Now that the integration on the Okta side is complete, it's time to set up our Directus app. We need to open the _.env_ file and add the following variables to it:

Reviewer: ```suggestion
Now that the integration on the Okta side is complete, it's time to set up our Directus app. We need to open the `.env` file and add the following variables to it:
```

| `AUTH_OKTA_ALLOW_PUBLIC_REGISTRATION` | true                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | Whether our application will register users who log in via SSO for the first time.                                                                                                                                                                                                                                               |

> | `AUTH_OKTA_DEFAULT_ROLE_ID`           | {Look in the database}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | Id of the role in the application that will be assigned by default to users logging in through Okta. We can look it up in the database. Note that the id can be different if we use different databases for development and production.                                                                                          |

Reviewer: ```suggestion
| `AUTH_OKTA_DEFAULT_ROLE_ID`           | {Look in the database}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | ID of the role in the application that will be assigned by default to users logging in through Okta. We can look it up in the database.                                                            |
```

True but also a given, in my opinion.

<md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="https://trial-3528890.okta.com/app/trial-3528890_mydirectusapp_1/exk8y5vtf9CddTPJ4697/sso/saml"/>
<md:SingleSignOnService Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect" Location="https://trial-3528890.okta.com/app/trial-3528890_mydirectusapp_1/exk8y5vtf9CddTPJ4697/sso/saml"/>
</md:IDPSSODescriptor>
</md:EntityDescriptor>'

AUTH_OKTA_SP_metadata='<EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:ds="http://www.w3.org/2000/09/xmldsig#" entityID="http://www.okta.com/exk8y5vtf9CddTPJ4697">
                       <SPSSODescriptor WantAssertionsSigned="true" protocolSupportEnumeration="urn:oasis:names:tc:SAML:2.0:protocol">
                       <NameIDFormat>urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress</NameIDFormat>
                       <AssertionConsumerService isDefault="true" index="0" Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location="http://localhost:8055/auth/login/okta/acs"/>
                       </SPSSODescriptor>
                       </EntityDescriptor>'
...
```

## Testing

> Now, let's make sure that the whole configuration works. Launch our Directus application, go to http://localhost:8055/ in the browser, and immediately see the change - we have a **Log In with Okta** button.

Reviewer: ```suggestion
Launch the Directus project, go to http://localhost:8055/ in the browser, and immediately see the change - we have a **Log In with Okta** button.
```

Using project here instead of application as we also use Okta 'app'.

![Application login screen with Log In with Okta button](app-login-screen.png)

Clicking on it takes us to the Okta login screen. 

> ![Okta login screen](okta-login-screen.png)

Reviewer: We can remove this. Most people know what this screen would look like.
